src/webui/www/private/scripts/client.js

```diff
@@ -544,8 +579,74 @@ window.addEvent('load', function() {
         processServerState();
     });
 
+    $('showSearchEngineLink').addEvent('click', function(e) {
+        showSearchEngine = !showSearchEngine;
+        localStorage.setItem('show_search_engine', showSearchEngine.toString());
+        if (showSearchEngine) {
+            $('showSearchEngineLink').firstChild.style.opacity = '1';
+            $('mainWindowTabs').removeClass('invisible');
+
+            addMainWindowTabsEventListener();
+        }
+        else {
+            $('showSearchEngineLink').firstChild.style.opacity = '0';
+            $('mainWindowTabs').addClass('invisible');
+            $("transfersTabLink").click();
+
+            addMainWindowTabsEventListener(false);
+        }
+    });
+
     $('StatisticsLink').addEvent('click', StatisticsLinkFN);
 
+    // main window tabs
+
+    var showTransfersTab = function(show) {
+        if (show === false) {
+            $("filtersColumn").addClass("invisible");
+            $("filtersColumn_handle").addClass("invisible");
+            $("mainColumn").addClass("invisible");
+            MochaUI.Desktop.resizePanels();
+        }
+        else {
+            $("filtersColumn").removeClass("invisible");
+            $("filtersColumn_handle").removeClass("invisible");
+            $("mainColumn").removeClass("invisible");
+
+            customSyncMainDataInterval = undefined;
+            clearTimeout(syncMainDataTimer);
+            syncMainDataTimer = syncMainData.delay(100);
+
+            showSearchTab(false);
+        }
+    };
+
+    var showSearchTab = function(show) {
+        if (show === false) {
```

Comment:
I needed to distinguish between `false` and `undefined` so that not passing a parameter would default to true. But I'm just going to refactor this code to make it easier to read.